flocker/control/test/test_protocol.py

```diff
@@ -693,25 +717,99 @@ def test_configuration_change(self):
         A configuration change results in connected protocols being notified
         of new cluster status.
         """
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
         service = build_control_amp_service(self)
         service.startService()
-        protocol = ControlAMP(Clock(), service)
-        protocol.makeConnection(StringTransport())
-        sent = []
-        self.patch_call_remote(sent, protocol=protocol)
+        server = LoopbackAMPClient(client.locator)
+        service.connected(server)
 
         service.configuration_service.save(TEST_DEPLOYMENT)
-        # Should only be one callRemote call.
-        (sent,) = sent
-        self.assertArgsEqual(
-            sent,
-            (
-                (ClusterStatusCommand,),
-                dict(
-                    configuration=TEST_DEPLOYMENT,
-                    state=DeploymentState(),
-                )
-            )
+
+        self.assertEqual(
+            dict(configuration=TEST_DEPLOYMENT, state=DeploymentState()),
+            dict(configuration=agent.desired, state=agent.actual),
+        )
+
+    def test_second_configuration_change_waits_for_first_acknowledgement(self):
+        """
+        A second configuration change is only transmitted after acknowledgement
+        of the first configuration change is received.
+        """
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
+        service = build_control_amp_service(self)
+        service.startService()
+
+        configuration = service.configuration_service.get()
+        modified_configuration = arbitrary_transformation(configuration)
+
+        server = LoopbackAMPClient(client.locator)
+        delayed_server = DelayedAMPClient(server)
+        # Send first update
+        service.connected(delayed_server)
```

Comment:
:question: This and the following test simulate the sending of state change when agent first connects to control service...which is special in that the first state is only sent to that agent, not all agents. Does that matter? I guess not...